src/rosconsole/impl/rosconsole_log4cxx.cpp

```diff
@@ -354,6 +358,15 @@ void register_appender(LogAppender* appender)
   logger->addAppender(g_log4cxx_appender);
 }
 
+void deregister_appender(LogAppender* appender){
+  if(g_log4cxx_appender->getAppender() == appender)
+  {
+    const log4cxx::LoggerPtr& logger = log4cxx::Logger::getLogger(ROSCONSOLE_ROOT_LOGGER_NAME);
+    logger->removeAppender(g_log4cxx_appender);
+    delete g_log4cxx_appender;
+    g_log4cxx_appender = 0;
```

Comment:
With the `g_log4cxx_appender` being deleted and set to zero I think the `shutdown` function should check if `g_log4cxx_appender` is not `0` before trying to remove it.

Also initializing the variable in line 352 with `0` might be good.